app/src/main/java/com/projectit3048c/MainActivity.kt

```diff
@@ -165,50 +165,46 @@ class MainActivity : ComponentActivity() {
             )
             Button(
                 onClick = {
-                    var specimen = FoodAmount().apply {
+                    var foodAmount = FoodAmount().apply {
```

Comment:
The variable name "specimen" should be renamed to something more appropriate for this domain, but... note that foodAmount is already used in another location; within the apply{} block. Thus you have a shadow condition: two variables with the same name.  So, I'd rename it, but to something else.